design-proposals/memory-hotplug.md

```diff
@@ -0,0 +1,209 @@
+under `spec.domain.memory` in the VMI object. However, all VMI Memory fields will stay immutable, and Memory guest will
+only be adjustable on the VM object. Once the hoplug action begins, subsequent changes to the number of guest would be
+rejected until the hotplug process completes. Memory Resource update on the pod level must occur before the actual
+hotplug action on the Libvirt level, hence, during the hotplug action a notifitcation will be created upon successful
+update of the Memory resources on the pod level. It will indicate that now it safe to proceed to the hotplug action on
+the Libvirt/QEMU level.
+
+## Drawbacks / Linmitations
+
+* Workload update method should be defined explicitly as `VMLiveUpdateFeatures` under the Kubevirt configuration.
+* This feature will be incompatible with memory requests/limits, at least initially, since those values need to
+  dynamically scale according to the current number of enabled memory
+* Set the memory increment (new_guest - old_guest) for online memory expansion An integer multiple of 128 MB (operating
+  system requirements)
+* Memory hotplug depends on numa, VMs will have to be `NUMAable` to enable Memory hotplug
+
+#### API Examples
+
+##### VM Spec API
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-cirros
+  name: vm-cirros
+spec:
+  running: false
+  liveUpdateFeatures:
+  +   memory
++     maximumGuest: 4Gi
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-cirros
+    spec:
+      domain:
+        memory:
+          guest: 4Gi
+```
+
+##### VMI Spec API
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  labels:
+    special: vm-cirros
+  name: vm-cirros
+spec:
+  domain:
+    memory:
++     maxGuest: 10Gi
+  guest: 8Gi
+
+```
+
+## Status API
+
+The VMI status must indicate whether Memory hotplug is taking place. As the hotplug process begins the current state
+of `spec.domain.memory` needs to be propagated back to the VM/VMI `status.currentMemory` to allow the user to have an
+indication of the hotplug processing.
+
+```
+	// Memory allow specified the detailed inside the vmi.
+		CurrentMemory *Memory `json:"currentMemory,omitempty"`
+```
+
+Conditions on the VM and VMI should indicate the status of the pending Memory hotplug actions. When errors occur, the
+conditions should propagate the errors to the VM.Status.Conditions field so users can receive feedback into the status
+of their declared state.
+
+```
+// Indicates that the VMI is in progress of Hot memory Plug/UnPlug
+VirtualMachineInstanceMemoryChange VirtualMachineInstanceConditionType = "HotMemoryChange"
+```
+
+Upon succesful hotplug action the condition will be removed. Otherwise, a human readable reason and message indicating
+whether the hotplug change is pending or encountering errors.
+
+## Extensibility
+
+We will be able to use the `VMLiveUpdateFeatures` section to manage things like cpu/memory hotplug .
```

Comment:
You can remove this section, it was already mentioned in the CPU hotplug design where the `LiveUpdateFeatures` API was introduced.

Reply:
ok